### PR TITLE
Handle thunder id based on platform type

### PR DESCRIPTION
**Pull request by HaseenaSainul**

### CODE REVIEW

File: Source/core/JSON.h [hunk 1]
```diff
@@ -5028,6 +5027,112 @@ namespace Core {
...
+            }
+
+            ThreadId(const ThreadId& copy)
+                : String(copy)
+            {
+            }
+
+            ~ThreadId() override = default;
+
+            ThreadId& operator=(const ThreadId&& move)
```
Comment (sebaszm) [inline Source/core/JSON.h]: const on moved param

File: Tests/unit/core/test_workerpool.cpp [hunk 2]
```diff
@@ -260,10 +260,10 @@ namespace Core {
             const uint8_t MaxSize = 15;
             bool isPoolId = false;
             char id[MaxSize];
-            sprintf(id, "%x", static_cast<::ThreadId>(pthread_self()));
+            sprintf(id, "%x", static_cast<::thread_id>(pthread_self()));
```
Comment (sebaszm) [inline Tests/unit/core/test_workerpool.cpp]: ::Thunder::Core::thread_id

File: Source/Thunder/Controller.cpp [hunk 3]
```diff
@@ -1262,7 +1262,7 @@ namespace Plugin {
 
             while (it.Next() == true) {
                 auto const& entry = it.Current();
-                threads.push_back({ entry.Id.Value(), entry.Job.Value(), entry.Runs.Value() });
+                threads.push_back({ Id(entry.Id.Value()), entry.Job.Value(), entry.Runs.Value() });
```
Comment (sebaszm) [inline Source/Thunder/Controller.cpp]: Doesn't JSON::ThreadId loose its purpose now? Likewise the parallel changes to JsonGenerator.

**Review by sebaszm — CHANGES_REQUESTED** [review]

File: Tests/unit/core/test_workerpool.cpp (hunk 2 above)
Comment (HaseenaSainul) [reply Tests/unit/core/test_workerpool.cpp]: done

File: Source/core/JSON.h (hunk 1 above)
Comment (HaseenaSainul) [reply Source/core/JSON.h]: done

File: Source/Thunder/Controller.cpp (hunk 3 above)
Comment (HaseenaSainul) [reply Source/Thunder/Controller.cpp]: done

**Review by sebaszm — APPROVED** [review]

**Review by MFransen69 — APPROVED** [review]

**Pull request merged**